src/SDKs/Azure.Core/data-plane/Azure.Core/Pipeline/Policies/RetryPolicy.cs

```diff
@@ -85,18 +104,30 @@ public override async Task ProcessAsync(HttpPipelineMessage message, ReadOnlyMem
 
                 if (delay > TimeSpan.Zero)
                 {
-                    await DelayAsync(delay, message.Cancellation);
+                    if (async)
+                    {
+                        await WaitAsync(delay, message.Cancellation);
+                    }
+                    else
+                    {
+                        Wait(delay, message.Cancellation);
+                    }
                 }
 
                 HttpPipelineEventSource.Singleton.RequestRetrying(message.Request, attempt);
             }
         }
 
-        internal virtual async Task DelayAsync(TimeSpan time, CancellationToken cancellationToken)
+        internal virtual async Task WaitAsync(TimeSpan time, CancellationToken cancellationToken)
```

Comment:
Should we consider keeping the name as `DelayAsync` to avoid confusion with the blocking `Wait` semantics for tasks?  While I'd agree that "wait" is a more accurate verb, I think the overloaded meaning makes it less clear in this context.

Reply:
Both fixed and exponential retry policy have Delay property on them, that's why I had to rename the method to be able to define `Wait` sync overload.